src/battle_util.c

```diff
@@ -9393,270 +9375,409 @@ static u32 CalcDefenseStat(u16 move, u8 battlerAtk, u8 battlerDef, u8 moveType,
     switch (holdEffectAtk)
     {
     case HOLD_EFFECT_METRONOME:
         percentBoost = min((gBattleStruct->sameMoveTurns[battlerAtk] * GetBattlerHoldEffectParam(battlerAtk)), 100);
-        finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(1.0) + sPercentToModifier[percentBoost]);
+        return sPercentToModifier[percentBoost];
         break;
     case HOLD_EFFECT_EXPERT_BELT:
         if (typeEffectivenessModifier >= UQ_4_12(2.0))
-            finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(1.2));
+            return UQ_4_12(1.2);
         break;
     case HOLD_EFFECT_LIFE_ORB:
-        finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(1.3));
+        return UQ_4_12(1.3);
         break;
     }
+    return UQ_4_12(1.0);
+}
+
+static inline uq4_12_t GetDefenderItemsModifier(u32 moveType, u32 battlerDef, uq4_12_t typeEffectivenessModifier, bool32 updateFlags)
+{
+    u32 holdEffectDef = GetBattlerHoldEffect(battlerDef, TRUE);
+    u32 holdEffectDefParam = GetBattlerHoldEffectParam(battlerDef);
+    u32 itemDef = gBattleMons[battlerDef].item;
+    u32 abilityDef = GetBattlerAbility(battlerDef);
 
-    // target's hold effect
     switch (holdEffectDef)
     {
-    // berries reducing dmg
     case HOLD_EFFECT_RESIST_BERRY:
-        if (moveType == GetBattlerHoldEffectParam(battlerDef)
-            && (moveType == TYPE_NORMAL || typeEffectivenessModifier >= UQ_4_12(2.0))
-            && !UnnerveOn(battlerDef, itemDef))
+        if (UnnerveOn(battlerDef, itemDef))
+            return UQ_4_12(1.0);
+        if (moveType == holdEffectDefParam && (moveType == TYPE_NORMAL || typeEffectivenessModifier >= UQ_4_12(2.0)))
         {
-            if (abilityDef == ABILITY_RIPEN)
-                finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(0.25));
-            else
-                finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(0.5));
             if (updateFlags)
                 gSpecialStatuses[battlerDef].berryReduced = TRUE;
+            return (abilityDef == ABILITY_RIPEN) ? UQ_4_12(0.25) : UQ_4_12(0.5);
         }
         break;
     }
+    return UQ_4_12(1.0);
+}
 
-    if (gBattleMoves[move].minimizeDoubleDamage         && gStatuses3[battlerDef] & STATUS3_MINIMIZED)
-        finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(2.0));
-    if (gBattleMoves[move].damagesUnderground           && gStatuses3[battlerDef] & STATUS3_UNDERGROUND)
-        finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(2.0));
-    if (gBattleMoves[move].damagesUnderwater            && gStatuses3[battlerDef] & STATUS3_UNDERWATER)
-        finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(2.0));
-    if (gBattleMoves[move].damagesAirborneDoubleDamage  && gStatuses3[battlerDef] & STATUS3_ON_AIR)
-        finalModifier = uq4_12_multiply(finalModifier, UQ_4_12(2.0));
-
-    dmg = ApplyModifier(finalModifier, dmg);
-    if (dmg == 0)
-        dmg = 1;
+#define DAMAGE_ACCUMULATE_MULTIPLIER(modifier) do {                     \
```

Comment:
Maybe `APPLY_DMG_MULTIPLIER` would be less verbose?

Reply:
I'm not stuck to thes names by any means :D I tired to convey the following:

* `DAMAGE_APPLY_MODIFIER` will multiply an integer by a FP number and round (with gf rounding)
* `DAMAGE_ACCUMULATE_MULTIPLIER` will multiply a FP number by a FP number and round (with gf rounding)

the `apply` at least in my head signals this, while `accumulate` say we still have a FP number in the end.

Now that I think about it that is probably not clear to everyone, I'm open for suggestions :)

Reply:
we use `MulModifier` for type effectiveness, so `DAMAGE_MUL_MULTIPLIER` might be more intuitive

Reply:
ah given that the variable is called `finalModifier` (which I dont object to) using `_MODIFIER` for the macros makes sense. 👍

Reply:
I chose `DAMAGE_APPLY_MODIFIER` and `DAMAGE_MULTIPLY_MODIFIER` respectively :)